### DISCUSSION TITLE
feat: better context management

### DISCUSSION THREAD

**Discussion by eureka-0** [open-webui/open-webui#6741] [General]
# Feature Request

**Is your feature request related to a problem? Please describe.**
As an OpenAI API user, I often don't need much context (usually 4 or fewer), since longer contexts increase the cost. This situation likely applies to most of OpenAI or Claude API users. Therefore, implementing built-in context management would be beneficial, with good examples like LobeChat providing inspiration.

**Describe the solution you'd like**
A built-in context management based on conversations, and allow me to change its value at any time.

**Describe alternatives you've considered**
I tried using the "context clip filter" function, but found it inconvenient. To use this function, I have to create a new model specifically incorporating it. Additionally, if I change the history message value of this function, it affects all ongoing conversations that use the model. This makes it difficult to manage multiple conversations with different history message values.

**Additional context**
Take LobeChat's context management capabilities as an example:

<img width="322" alt="image" src="https://github.com/user-attachments/assets/404ffc25-b55a-4308-a0a1-e9c21f28e241">

<img width="296" alt="image" src="https://github.com/user-attachments/assets/c6bce917-97e9-4d13-b690-c58d2c459be0">

**Comment by tjbck** [comment]
I'd greatly appreciate if you could just do a quick search.

https://openwebui.com/f/hub/context_clip_filter/

**Reply by eureka-0** [reply]
> I'd greatly appreciate if you could just do a quick search.
> 
> https://openwebui.com/f/hub/context_clip_filter/

Thank you for reply. I tried using this function, but found it inconvenient. To use this function, I have to create a new model specifically incorporating it. Additionally, if I change the history message value of this function, it affects all ongoing conversations that use the model. This makes it difficult to manage multiple conversations with different history message values.

**Reply by tjbck** [reply]
Understandable, will think of ways to add native support for this feature. Stay tuned!
